http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskV2ServiceInterp.scala

```diff
@@ -115,6 +97,27 @@ class DiskV2ServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
 
       _ <- ctx.span.traverse(s => F.delay(s.addAnnotation("Done auth call for delete azure disk permission")))
 
+      // check that workspaceId is not null
+      workspaceId <- F.fromOption(disk.workspaceId, DiskWithoutWorkspaceException(diskId, ctx.traceId))
+
+      hasWorkspacePermission <- authProvider.isUserWorkspaceReader(
+        WorkspaceResourceSamResourceId(workspaceId),
+        userInfo
+      )
+      _ <- F.raiseUnless(hasWorkspacePermission)(ForbiddenError(userInfo.userEmail))
+
+      // check if disk resource is deletable in WSM
+      wsmResourceId <- F.fromOption(disk.wsmResourceId, DiskWithoutWsmResourceIdException(disk.id, ctx.traceId))
+      wsmStatus <- wsmClientProvider.getDiskState(userInfo.accessToken.token, workspaceId, wsmResourceId)
+
+      _ <- F.raiseUnless(wsmStatus.isDeletable)(
+        DiskCannotBeDeletedWsmException(disk.id, wsmStatus, disk.cloudContext, ctx.traceId)
+      )
```

Comment:
... as the caller asked to delete the disk, should we go through some level of either explanation back to the user or attempts on our end to delete the disk?

Reply:
I added a message to the error here for the user to wait and retry because the non-deletable messages are in-progress ones (CREATING, UPDATING, DELETING)